### ISSUE TITLE
Some SB devices don't play on 0.4.0 Free Account

### ISSUE THREAD

**Issue by Archcantor** [sspiff/lms-plugin-pyrrha#29]
Thanks for the free account update!  This plugin works on Squeezelite and on my SB Baby Radios. It does not work on my Duets nor my Boom Radio.  The display on the Boom returns "Problem: Unable to play file"  The Boom is on firmware 57.  The Duets are on Firmware 77.  The working Baby Radios are on Firmware: 7.7.3-r16676

**Comment by sspiff** [comment]
> Thanks for the free account update! This plugin works on Squeezelite and on my SB Baby Radios. It does not work on my Duets nor my Boom Radio. The display on the Boom returns "Problem: Unable to play file" The Boom is on firmware 57. The Duets are on Firmware 77. The working Baby Radios are on Firmware: 7.7.3-r16676

Anything interesting in the LMS log?

**Comment by sspiff** [comment]
Similar issue reported on the forum: https://forums.slimdevices.com/forum/user-forums/3rd-party-software/1675996-pyrrha-pandora-s-daughter?p=1683756#post1683756

**Comment by sspiff** [comment]
Two instances of this reported on the forum, and both appear to involve LMS on Windows.

@Archcantor , are you seeing this with LMS on Windows?

**Comment by Archcantor** [comment]
> Two instances of this reported on the forum, and both appear to involve LMS on Windows.
> 
> @Archcantor , are you seeing this with LMS on Windows?

Yes, LMS on Windows 
Logitech Media Server Version: 8.4.1 - 1709721691

**Comment by Archcantor** [comment]
> > Thanks for the free account update! This plugin works on Squeezelite and on my SB Baby Radios. It does not work on my Duets nor my Boom Radio. The display on the Boom returns "Problem: Unable to play file" The Boom is on firmware 57. The Duets are on Firmware 77. The working Baby Radios are on Firmware: 7.7.3-r16676
> 
> Anything interesting in the LMS log?

LMS on Windows 11
Logitech Media Server Version: 8.4.1 - 1709721691

FROM THE BOOM RADIO THAT WON'T PLAY
[24-03-08 18:58:40.0590] Plugins::Pyrrha::Pandora::__ANON__ (137) fetching station list
[24-03-08 18:58:40.0605] Plugins::Pyrrha::Pandora::__ANON__ (84) creating new websvc
[24-03-08 18:58:43.9976] Plugins::Pyrrha::ProtocolHandler::getNextTrack (144) pyrrha://c529fc275c549e469d68781ca9ef46ed/4650233934030208209.mp3
[24-03-08 18:58:43.9977] Plugins::Pyrrha::ProtocolHandler::getNextTrack (171) found cached station data 68904770220752081
[24-03-08 18:58:43.9978] Plugins::Pyrrha::ProtocolHandler::getNextTrack (172) playlist length: 3
[24-03-08 18:58:43.9978] Plugins::Pyrrha::ProtocolHandler::getNextTrack (173) station change 4650233934030208209
[24-03-08 18:58:45.7116] Plugins::Pyrrha::ProtocolHandler::__ANON__ (211) next in playlist: ca0a6eed47dd480f2e2c4f404959b97a
[24-03-08 18:58:45.7120] Slim::Player::Song::open (424) Error: Couldn't create command line for mp4 playback for [pyrrha://c529fc275c549e469d68781ca9ef46ed/4650233934030208209.mp3]
[24-03-08 18:58:45.7129] Plugins::Pyrrha::ProtocolHandler::getNextTrack (144) pyrrha://c529fc275c549e469d68781ca9ef46ed/4650233934030208209.mp3
[24-03-08 18:58:45.7130] Plugins::Pyrrha::ProtocolHandler::getNextTrack (171) found cached station data 4650233934030208209
[24-03-08 18:58:45.7130] Plugins::Pyrrha::ProtocolHandler::getNextTrack (172) playlist length: 4
[24-03-08 18:58:45.7134] Plugins::Pyrrha::ProtocolHandler::getNextTrack (144) pyrrha://c529fc275c549e469d68781ca9ef46ed/4650233934030208209.mp3
[24-03-08 18:58:45.7135] Plugins::Pyrrha::ProtocolHandler::getNextTrack (171) found cached station data 4650233934030208209
[24-03-08 18:58:45.7135] Plugins::Pyrrha::ProtocolHandler::getNextTrack (172) playlist length: 4
[24-03-08 18:58:45.7740] Plugins::Pyrrha::ProtocolHandler::__ANON__ (211) next in playlist: 0529a720fbfc2ade7c3e534a0527998a
[24-03-08 18:58:45.7743] Plugins::Pyrrha::ProtocolHandler::__ANON__ (211) next in playlist: 4650233934030208209-pod:1/1/0
[24-03-08 18:58:45.7749] Slim::Player::Song::open (424) Error: Couldn't create command line for mp4 playback for [pyrrha://c529fc275c549e469d68781ca9ef46ed/4650233934030208209.mp3]

FROM THE BABY RADIO THAT PLAYS
[24-03-08 19:02:24.2891] Plugins::Pyrrha::ProtocolHandler::getNextTrack (144) pyrrha://c529fc275c549e469d68781ca9ef46ed/4650233934030208209.mp3
[24-03-08 19:02:24.2892] Plugins::Pyrrha::ProtocolHandler::getNextTrack (171) found cached station data 4650233934030208209
[24-03-08 19:02:24.2893] Plugins::Pyrrha::ProtocolHandler::getNextTrack (172) playlist length: 4
[24-03-08 19:02:24.2922] Plugins::Pyrrha::ProtocolHandler::__ANON__ (211) next in playlist: a5b607c02d1a719298493fefad10e051

**Comment by sspiff** [comment]
> > Two instances of this reported on the forum, and both appear to involve LMS on Windows.
> > @Archcantor , are you seeing this with LMS on Windows?
> 
> Yes, LMS on Windows Logitech Media Server Version: 8.4.1 - 1709721691

Do you know if you are able to play mp4/aac streams from other sources?

In the past, it looks like an extra step was required to enable mp4/aac transcoding on Windows.  See https://wiki.slimdevices.com/index.php/AAC.html (but I don't know how old this is or if the info is still valid).

**Comment by Archcantor** [comment]
> > > Two instances of this reported on the forum, and both appear to involve LMS on Windows.
> > > @Archcantor , are you seeing this with LMS on Windows?
> > 
> > 
> > Yes, LMS on Windows Logitech Media Server Version: 8.4.1 - 1709721691
> 
> Do you know if you are able to play mp4/aac streams from other sources?
> 
> In the past, it looks like an extra step was required to enable mp4/aac transcoding on Windows. See https://wiki.slimdevices.com/index.php/AAC.html (but I don't know how old this is or if the info is still valid).

The Boom plays an MP4 file from the local drive.  Squeezelite and Baby do not play the MP4 file.

The Boom and Baby play this stream: https://ais-sa5.cdnstream1.com/2133_64.aac.  Squeezelite runs the clock but no audio.

**Comment by sspiff** [comment]
> FROM THE BOOM RADIO THAT WON'T PLAY

> FROM THE BABY RADIO THAT PLAYS

> > Do you know if you are able to play mp4/aac streams from other sources?
> 
> The Boom plays an MP4 file from the local drive. Squeezelite and Baby do not play the MP4 file.
> 
> The Boom and Baby play this stream: https://ais-sa5.cdnstream1.com/2133_64.aac. Squeezelite runs the clock but no audio.

So... the Boom will play a local MP4 file, but not the MP4 stream from Pyrrha?  While the baby will play the MP4 stream from Pyrrha, but not the local MP4 file?

**Comment by Archcantor** [comment]
> > FROM THE BOOM RADIO THAT WON'T PLAY
> 
> > FROM THE BABY RADIO THAT PLAYS
> 
> > > Do you know if you are able to play mp4/aac streams from other sources?
> > 
> > 
> > The Boom plays an MP4 file from the local drive. Squeezelite and Baby do not play the MP4 file.
> > The Boom and Baby play this stream: https://ais-sa5.cdnstream1.com/2133_64.aac. Squeezelite runs the clock but no audio.
> 
> So... the Boom will play a local MP4 file, but not the MP4 stream from Pyrrha? While the baby will play the MP4 stream from Pyrrha, but not the local MP4 file?

Correct.  I also installed QuickTime per your link and confirmed Lame is enabled.  No change. The latest QuickTime is old from 2016. I should mention LMS is the 64 bit Dev version.

**Comment by sspiff** [comment]
If I set my squeezelite player to only accept mp3, then I get similar "Couldn't create command line" errors on LMS on Linux.

**Comment by AnonymousLMSUser** [comment]
"Couldn't create command line for mp4" error is not specific to Windows.  I can confirm consistent behavior with older hardware connected to a linux-based LMS.  However, a functional workaround already exists thanks to forum user bpa and the "Play MP4 files & streams" plugin.  To add to LMS, add the following third-party plugin URL to Additional Repositories under Manage Plugins:
http://downloads.sourceforge.net/project/bpaplugins/repo-playmp4.xml

**Comment by Archcantor** [comment]
> http://downloads.sourceforge.net/project/bpaplugins/repo-playmp4.xml

In case this helped with LMS in Windows I installed this plugin.  No Change.  The Boom and Duets do not play the free Pandora stream.  The display on the Boom returns "Problem: Unable to play file"  Starting the Boom stream with Squeezectrl on Android returns "Unable to play file type for: [stream name]

Squeezelite and Baby radios play fine.

**Comment by sspiff** [comment]
I think I've got a fix in the works.  Need to do some more testing, but seems promising.

**Comment by Archcantor** [comment]
I installed 0.4.1 and it still does not play the free account on the Boom or Duets.  Different log messages.

The Boom does not display any error. It plays silence for about 3 seconds and stops.  Squeezelite and Baby play fine.

My LMS server was the x64 experimental version 8.4.1  
I installed 8.5.0 latest 32 bit version but no change.

Thanks for your work on this!

[24-03-14 20:47:16.9756] Slim::Networking::IO::Select::__ANON__ (130) Error: Select task failed calling Slim::Web::HTTP::processHTTP: Not a SCALAR reference at C:/Program Files/Squeezebox/server/Slim/Web/HTTP.pm line 1828.
; fh=Slim::Web::HTTP::ClientConn=GLOB(0xed7bd20)
[24-03-14 20:47:19.5372] Plugins::Pyrrha::ProtocolHandler::getNextTrack (142) pyrrha://c529fc275c549e469d68781ca9ef46ed/4650233934030208209.mp3
[24-03-14 20:47:19.5373] Plugins::Pyrrha::ProtocolHandler::getNextTrack (169) found cached station data 4650233934030208209
[24-03-14 20:47:19.5374] Plugins::Pyrrha::ProtocolHandler::getNextTrack (170) playlist length: 5
[24-03-14 20:47:19.5399] Plugins::Pyrrha::ProtocolHandler::__ANON__ (209) next in playlist: c10e3fdf9789918cf0455809c0e2498d
[24-03-14 20:47:19.6464] Plugins::Pyrrha::ProtocolHandler::new (29) PH:new(): http://audio-dc6-t1-2-v4v6.pandora.com/access/4561543219812847789.mp4?version=5&lid=5668049&token=olV7RDMWV19mI%2Br4NhvXr3MK45sZ3GbJO2xF6IsLYlVdjsQoeG2xIpa5FSNCaInlA4tRQrIYcNu2p00VgTkzjFm8jAUPPrssOMGeQm8jZJpcyVl47InY8TT%2FihobkeT8iOwzg%2Fh5mcouRUc6DWjUgBQfUQ4sTm6JVzuSP7%2FEo5aq7ex%2B3NbRYVy0eMRmIQBMz6upRe3WGUNGu%2FEUOUuSwpN9%2BwHrF8w25fayNp3OwiL5QzcCklaLQGM3AujayFuczpkl1%2FKi%2FiQt8xU7x2iflYoEEvKRsSmi6ZLSB%2FGDx3JQCgFx0ZvLfCG5Ws28feJiw1xTtorCrx4%3D
[24-03-14 20:47:20.5500] Plugins::Pyrrha::ProtocolHandler::getNextTrack (142) pyrrha://c529fc275c549e469d68781ca9ef46ed/4650233934030208209.mp3
[24-03-14 20:47:20.5501] Plugins::Pyrrha::ProtocolHandler::getNextTrack (169) found cached station data 4650233934030208209
[24-03-14 20:47:20.5501] Plugins::Pyrrha::ProtocolHandler::getNextTrack (170) playlist length: 4
[24-03-14 20:47:20.5745] Plugins::Pyrrha::ProtocolHandler::__ANON__ (209) next in playlist: 930bd915feaf82dbaae303a75aed062d
[24-03-14 20:47:20.7591] Plugins::Pyrrha::ProtocolHandler::new (29) PH:new(): http://audio-sv5-t1-1-v4v6.pandora.com/access/8412054513448616817.mp4?version=5&lid=5668049&token=olV7RDMWV19sgv8kzUE6FzO6Fg1vEjo%2FqfqT80F36zTwux0%2B6zY%2BhYaZV5jVp5ayEfbJL%2BXmhd5VFCXWyjkCi6pCuH5hU3goCwqTcwYgEDtawsQSyE9cafnQ7WTccVPNb%2BZN%2FeRbJY9u7hvTWHQc6J59lJAgel77DVNnHPysDdwc8dUWPfl4G6OgnzxEuGFe4Rp0nCcy3ExJ7d8ezwxndy%2FXcd4OThpaICFz8LKF5Wafbptu%2FN5hn%2BNG3Bv5NeGDacYLuBOIuGLxbRsbfsNUklP%2BCf5IyQ54bLItEsMNp%2BMmk3Gpa%2BWmkhImZsp5qausCq2ZRPPPH6y0zgrILxF7xQ%3D%3D
